src/generic-site/content-script.js

```diff
@@ -34,5 +35,38 @@ function updateSettings(new_settings = null) {
     }
 }
 chrome.runtime.onMessage.addListener(handleMessage);
+
+function sendRollToGameLog(message) {
+    console.log(message)
+    chrome.runtime.sendMessage(message)
+}
+
+function disconnectAllEvents() {
+    registered_events.forEach((event) => {
+        event.disconnect();
+    })
+}
+
+function handleIncomingSettings({ character, action, type} = {}) {
+    characterInternal = character;
+    chrome.runtime.sendMessage({
+        action: "hp-update",
+        character
+    })
```

Comment:
incoming settings should send an hp-update. I also don't think a settings event usually includes a character 🤔